src/asm/aarch64/transform/forward.rs

```diff
@@ -0,0 +1,452 @@
+  );
+  into[0] = I32X8::new(stage2.0);
+  into[1] = I32X8::new(stage2.1);
+  into[2] = I32X8::new(int32x4x2_t(stage2.0 .1, stage2.0 .0));
+  into[3] = I32X8::new(int32x4x2_t(stage2.1 .1, stage2.1 .0));
+  into[4] = I32X8::new(stage2.2);
+  into[5] = I32X8::new(stage2.3);
+  into[6] = I32X8::new(int32x4x2_t(stage2.2 .1, stage2.2 .0));
+  into[7] = I32X8::new(int32x4x2_t(stage2.3 .1, stage2.3 .0));
+}
+
+#[inline]
+unsafe fn transpose_4x4_neon(input: &[I32X8; 4], into: &mut [I32X8; 4]) {
+  let stage1 = (
+    vtrnq_s32(input[0].vec().0, input[1].vec().0),
+    vtrnq_s32(input[2].vec().0, input[3].vec().0),
+  );
+  let stage2 = (
+    vtrnq_s64_to_s32(stage1.0 .0, stage1.1 .0),
+    vtrnq_s64_to_s32(stage1.0 .1, stage1.1 .1),
+  );
+  into[0] = I32X8::new(stage2.0);
+  into[1] = I32X8::new(stage2.1);
+  into[2] = I32X8::new(int32x4x2_t(stage2.0 .1, stage2.0 .0));
+  into[3] = I32X8::new(int32x4x2_t(stage2.1 .1, stage2.1 .0));
+}
+
+#[inline]
+unsafe fn shift_left_neon(a: I32X8, shift: u8) -> I32X8 {
+  let shift = vdupq_n_s32(shift.into());
+  I32X8::new(int32x4x2_t(
+    vrshlq_s32(a.vec().0, shift),
+    vrshlq_s32(a.vec().1, shift),
+  ))
+}
+
+#[inline]
+unsafe fn shift_right_neon<const SHIFT: i32>(a: I32X8) -> I32X8 {
+  I32X8::new(int32x4x2_t(
+    vrshrq_n_s32(a.vec().0, SHIFT),
+    vrshrq_n_s32(a.vec().1, SHIFT),
+  ))
+}
+
+#[inline]
+unsafe fn round_shift_array_neon(arr: &mut [I32X8], bit: i8) {
+  if arr.len() % 4 != 0 {
+    debug_unreachable!();
+  }
+
+  if bit == 0 {
+    return;
+  }
+  if bit > 0 {
+    if bit == 1 {
+      for s in arr.chunks_exact_mut(4) {
+        for chunk in s {
+          *chunk = shift_right_neon::<1>(*chunk)
+        }
+      }
+    } else if bit == 2 {
+      for s in arr.chunks_exact_mut(4) {
+        for chunk in s {
+          *chunk = shift_right_neon::<2>(*chunk)
+        }
+      }
+    } else {
+      debug_unreachable!();
+    }
+  } else {
+    let shift = (-bit) as u8;
+    for s in arr.chunks_exact_mut(4) {
+      for chunk in s {
+        *chunk = shift_left_neon(*chunk, shift);
+      }
+    }
+  }
+}
+
+#[allow(clippy::identity_op, clippy::erasing_op)]
+unsafe fn forward_transform_neon<T: Coefficient>(
+  input: &[i16], output: &mut [MaybeUninit<T>], stride: usize,
+  tx_size: TxSize, tx_type: TxType, bd: usize,
+) {
+  // Note when assigning txfm_size_col, we use the txfm_size from the
+  // row configuration and vice versa. This is intentionally done to
+  // accurately perform rectangular transforms. When the transform is
+  // rectangular, the number of columns will be the same as the
+  // txfm_size stored in the row cfg struct. It will make no difference
+  // for square transforms.
+  let txfm_size_col = tx_size.width();
+  let txfm_size_row = tx_size.height();
+
+  let col_class = SizeClass1D::from_length(txfm_size_col);
+  let row_class = SizeClass1D::from_length(txfm_size_row);
+
+  let mut tmp: Aligned<[I32X8; 64 * 64 / 8]> = Aligned::uninitialized();
```

Comment:
Put a note to reconsider this bit once the [maybeuninit methods for array](https://github.com/rust-lang/rust/issues/96097) hit stable.

Reply:
I went ahead and initialized the canonical way.
```rust
Aligned::new([MaybeUninit::<T>::uninit(); N])
```